### DISCUSSION TITLE
Pocketbase Stopped Logging (my solution)

### DISCUSSION THREAD

**Discussion by BobSnrub** [pocketbase/pocketbase#2152] [Q&A] [answered]
I had a weird problem that occurred around 3 months ago where my Pocketbase stopped logging.  Literally zero logs were made since January. I don't know what caused the problem but my dev environment and my build/serving environment both had the issue. I fixed it by stopping Pocketbase, deleting the `pb_data/logs.db`, and then restarting Pocketbase. It generated a new logs.db and everything worked again. Not sure if this is a "good" fix but I figured it was worth putting on here.

![image](https://user-images.githubusercontent.com/56611572/227600967-8543319e-6d30-4c61-9771-97b7f247ed62.png)

**Comment by ganigeorgiev** [comment]
I'm not sure how to reproduce it.

Are there any errors in the console if you start the executable with `--debug`?

What is the value of "Admin UI > Settings > Application > Logs max days retention"?

**Comment by BobSnrub** [comment, marked as answer]
Ah, yes. I reverted to the old logs.db and the problem still exists with that logs file.  The `--debug` revealed that the "`Log save failed: database disk image is malformed`".  I did have a power outage back in January where my pc shut off while Pocketbase was running so that could have done it I guess.  Anyway, starting with a new logs.db did fix the problem.

![image](https://user-images.githubusercontent.com/56611572/227620157-d8754854-616d-40bf-be1e-76b635dde8ee.png)

**Reply by ganigeorgiev** [reply]
Even in case of a unexpected server shutdown the logs db shouldn't be malformed. The most common reasons this could happen are:
- file permissions errors or permissions change while the application is running
- moving or deleting the db file while the application is running
- manually deleting the hot journal files (aka. the temp `-wal` files)

**Comment by discollizard** [comment]
quick update: as of jan 2025, logs are stored in the auxiliary.db file. i had to delete that instead
